src/core/geometry/qgsellipse.h

```diff
@@ -0,0 +1,219 @@
+/***************************************************************************
+                         qgsellipse.h
+                         --------------
+    begin                : March 2017
+    copyright            : (C) 2017 by Loîc Bartoletti
+    email                : lbartoletti at tuxfamily dot org
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSELLIPSE_H
+#define QGSELLIPSE_H
+
+#include <QString>
+
+#include "qgis_core.h"
+#include "qgspointv2.h"
+#include "qgspolygon.h"
+#include "qgslinestring.h"
+#include "qgsrectangle.h"
+
+/** \ingroup core
+ * \class QgsEllipse
+ * \brief Ellipse geometry type.
+ *
+ * An ellipse is defined by a center point (mCenter) with a semi-major axis, a semi-minor axis and an azimuth.
```

Comment:
I'd prefer to keep the internal variable names out of the API docs (since they aren't accessible through the python bindings)